### PR TITLE
properties relevant to dco:GeoSample

### PR DESCRIPTION
**Pull request by xgmachina**
Added and updated a few properties, by referring to the CSIRO IGSN metadata schema.

### CODE REVIEW

File: dco.ttl [hunk 1]
```diff
@@ -888,6 +889,31 @@ dco:IGSN
...
+      rdfs:domain dco:GeoSample ;
+      rdfs:label "sample type"@en-US ;
+      rdfs:comment  """
+      The type of a sample.
+      """^^rdf:XMLLiteral .
+
+dco:isPublic
+      a       owl:FunctionalProperty , owl:DatatypeProperty ;
+      rdfs:domain dco:GeoSample ;
+      rdfs:label "is public"@en-US ;
```
Comment (zednis) [inline dco.ttl:886]: Should this have a boolean range?

**Comment by xgmachina** [comment]
I was reading the subclass restrictions in the current definition of `dco:GeoSample`. Do they mean the required attributes of a geo-sample?

```
dco:GeoSample
      a       owl:Class ;
      rdfs:label "GeoSample"@en-US ;
      rdfs:subClassOf vivo:InformationResource ;
      ### May need to discuss the restrictions below if they mean to be the REQUIRED properties of a sample
      rdfs:subClassOf
              [ a       owl:Restriction ;
                owl:allValuesFrom rdfs:Literal ;
                owl:onProperty dco:hasElevation
              ] ;
      rdfs:subClassOf
              [ a       owl:Restriction ;
                owl:allValuesFrom rdfs:Literal ;
                owl:onProperty dco:hasMaterial
              ] ;
      rdfs:subClassOf
              [ a       owl:Restriction ;
                owl:allValuesFrom rdfs:Literal ;
                owl:onProperty dco:hasLongitude
              ] ;
      rdfs:subClassOf
              [ a       owl:Restriction ;
                owl:allValuesFrom rdfs:Literal ;
                owl:onProperty dco:hasLatitude
              ] ;
      rdfs:subClassOf
              [ a       owl:Restriction ;
                owl:allValuesFrom rdfs:Literal ;
                owl:onProperty dco:hasSampleAge
              ] ;
      rdfs:comment  """
      A specimen or small quantity of a geophysical material.
      """^^rdf:XMLLiteral .
```

**Comment by zednis** [comment]
@xgmachina that is the `owl:allValuesFrom` restriction which means that if the property exists it must have the type described in the restriction.  It does not mean the property must be present.

I do not think that is a useful class restriction in this case; it would be better to just provide a `rdfs:range` on the select properties.

**Comment by zednis** [comment]
for example:

```
dco:GeoSample
      a       owl:Class ;
      rdfs:label "GeoSample"@en-US ;
      rdfs:subClassOf vivo:InformationResource ;
      rdfs:subClassOf
              [ a       owl:Restriction ;
                owl:allValuesFrom rdfs:Literal ;
                owl:onProperty dco:hasElevation
              ] .
```

essentially means "If a `dco:Geosample` instance has property `dco:hasElevation` then the object of that property can be inferred to have type `rdfs:Literal`"

This is a useful class restriction if the type of the subject changes the expectation of the object type for a given property.

In this case though, I think we should just provide a range on the property and remove these restrictions.

**Comment by xgmachina** [comment]
Thanks @zednis I will made further updates on this. Perhaps will remove them.

**Comment by zednis** [comment]
If you want to assert that there is an elevation, material, longitude, latitude, etc for each sample you should use the `owl:someValuesFrom` restriction, but even then it is not a requirement but a statement that for each instance of type T the existence of property P is inferable.

http://www.w3.org/TR/owl-ref/#someValuesFrom-def:

> The owl:someValuesFrom constraint is analogous to the existential quantifier of Predicate logic - for each instance of the class that is being defined, there exists at least one value for P that fulfills the constraint.

**Comment by zednis** [comment]
Is this superceded by the current sample + sample repository efforts?

**Comment by mrpatrickwest** [comment]
I think so. I say we delete this branch and create a new one

**Comment by zednis** [comment]
I agree.  New branch.

**Pull request closed without merging**